controllers/front/success.php

```diff
@@ -47,19 +49,57 @@ public function postProcess()
 
             Tools::redirect($redirectLink);
         }
+        try {
+            $authResponseBody = $this->assertTransaction($orderId);
 
-        $orderLink = $this->context->link->getPageLink(
-            'order-confirmation',
-            true,
-            null,
-            [
-                'id_cart' => $cartId,
-                'id_module' => $moduleId,
-                'id_order' => $orderId,
-                'key' => $secureKey,
-            ]
-        );
+            if ($authResponseBody->getTransaction()->getStatus() === TransactionStatus::CANCELED) {
+                throw new Exception('Failed to authorize transaction');
```

Comment:
You don't need to throw exception here if you will catch it inside the same class. If status is cancelled, just log error inside condition braces.